cmd/nodecmd/whitelist.go

```diff
@@ -0,0 +1,170 @@
+		nodeConfig, err := app.LoadClusterNodeConfig(node)
+		if err != nil {
+			ux.Logger.PrintToUser("Failed to parse node %s due to %s", node, err.Error())
+		}
+		if slices.Contains(cloudSecurityGroupList, regionSecurityGroup{
+			cloud:         nodeConfig.CloudService,
+			region:        nodeConfig.Region,
+			securityGroup: nodeConfig.SecurityGroup,
+		}) {
+			continue
+		}
+		cloudSecurityGroupList = append(cloudSecurityGroupList, regionSecurityGroup{
+			cloud:         nodeConfig.CloudService,
+			region:        nodeConfig.Region,
+			securityGroup: nodeConfig.SecurityGroup,
+		})
+	}
+	if len(cloudSecurityGroupList) == 0 {
+		ux.Logger.PrintToUser("No nodes found in cluster %s", clusterName)
+		return nil
+	}
+
+	// GCP doesn't have regions  so we need to reduce it to only list of security groups
+	gcpSGList := []string{}
+	// whitelist IP
+	for _, cloudSecurityGroup := range cloudSecurityGroupList {
+		if cloudSecurityGroup.cloud == constants.GCPCloudService {
+			if !slices.Contains(gcpSGList, cloudSecurityGroup.securityGroup) {
+				gcpSGList = append(gcpSGList, cloudSecurityGroup.securityGroup)
+			}
+			continue
+		}
+		ux.Logger.PrintToUser("Whitelisting IP %s in %s cloud region %s", userIPAddress, cloudSecurityGroup.cloud, cloudSecurityGroup.region)
+		if cloudSecurityGroup.cloud == "" || cloudSecurityGroup.cloud == constants.AWSCloudService {
+			ec2Svc, err := awsAPI.NewAwsCloud(awsProfile, cloudSecurityGroup.region)
+			if err != nil {
+				ux.Logger.PrintToUser("Failed to establish connection to %s cloud region %s", cloudSecurityGroup.cloud, cloudSecurityGroup.region)
+				return err
+			}
+			securityGroupExists, sg, err := ec2Svc.CheckSecurityGroupExists(cloudSecurityGroup.securityGroup)
+			if err != nil || !securityGroupExists {
+				ux.Logger.PrintToUser("Can't find security group %s in %s cloud region %s", cloudSecurityGroup.securityGroup, cloudSecurityGroup.cloud, cloudSecurityGroup.region)
+				return err
+			}
+			ipInTCP := awsAPI.CheckUserIPInSg(&sg, userIPAddress, constants.SSHTCPPort)
+			ipInHTTP := awsAPI.CheckUserIPInSg(&sg, userIPAddress, constants.AvalanchegoAPIPort)
+			var alreadyWhitelisted bool
+			if ipInTCP {
+				alreadyWhitelisted = true
+			} else {
+				if err := ec2Svc.AddSecurityGroupRule(*sg.GroupId, "ingress", "tcp", userIPAddress, constants.SSHTCPPort); err != nil {
+					ux.Logger.PrintToUser("Failed to whitelist IP %s in %s cloud region %s for ssh access", userIPAddress, cloudSecurityGroup.cloud, cloudSecurityGroup.region)
+					return err
+				}
+			}
+			if ipInHTTP {
+				alreadyWhitelisted = true
+			} else {
+				if err := ec2Svc.AddSecurityGroupRule(*sg.GroupId, "ingress", "tcp", userIPAddress, constants.AvalanchegoAPIPort); err != nil {
+					ux.Logger.PrintToUser("Failed to whitelist IP %s in %s cloud region %s for http access", userIPAddress, cloudSecurityGroup.cloud, cloudSecurityGroup.region)
+					return err
+				}
+			}
+			if alreadyWhitelisted {
```

Comment:
this boolean may be set in two different cases, maybe we can to indicate if the already whitelisted situation refers to ssh or http

Reply:
fixed